### ISSUE TITLE
View label opens empty view with exception

### ISSUE THREAD

**Issue by rehand** [roelkneepkens/labelary-extension#5]
Hello,
when I try to press "view label" an empty view opens. I see the following exception:

```
Activation Event:onCommand:labelary.view-labeldefault
Activation Time:58ms
Uncaught Errors (3)
Request failed with status code 404
Request failed with status code 404
Request failed with status code 404
```

**Comment by roelkneepkens** [comment]
Dear Rehand,  is this issue still occurring?
Ifso: could you share the string you click on when this error happens?

**Comment by rehand** [comment]
There is no link to click on, but in the developer tools console I see the following exception:
```
 rejected promise not handled within 1 second: Error: Request failed with status code 404
p @ workbench.desktop.main.js:129
```

**Comment by bvanderhorn** [comment]
Hi @rehand , thank you for the information. 
Could you share with us your local file that contains your Base64 or ZPL string, or just the string itself, such that we can try and reproduce?

Thanks in advance!

**Comment by rehand** [comment]
I've removed some commands (including a Z64 image) from the whole file as it contains some company related things I'm not allowed to post here, but this file here doesn't work:
`^XA
^SZ2^JMA
^MCY^PMN
^PW1223~JSN
^JZY
^LH0,0^LRN
^FO29,476
^BY6^BCN,315,N,N^FD>;>8111222333444555^FS
^FT29,842
^A0N,50,69^FD111222333444555^FS
^PQ1,0,1,Y
^XZ
`

If I remove the "BY6" line then it works.

**Comment by roelkneepkens** [comment]
Hey @rehand,

If you select the full string it should give you the label (works for me).
If you only click on something inside the string it shows the empty page because it cannot determine the start and end of the ZPL string due to the `>` inside the string.

This is what we meant with this part the readme:
>In case the selection does not work: select string, right click and again select the option applicable.
![image](https://github.com/roelkneepkens/labelary-extension/assets/16499179/43e49890-362b-4a35-a05d-fac29f46a3a9)


If I remove the 2x `>` in the below string it seems to work:
`^XA ^SZ2^JMA ^MCY^PMN ^PW1223~JSN ^JZY ^LH0,0^LRN ^FO29,476 ^BY6^BCN,315,N,N^FD;8111222333444555^FS ^FT29,842 ^A0N,50,69^FD111222333444555^FS ^PQ1,0,1,Y ^XZ`

Can you confirm that the selection workaround works?

Ifso we can see if we can make the selector logic a bit smarter.

**Comment by rehand** [comment]
It doesn't work for me if I 2x remove the >.

In generale I'd have expected that "view label" without selection uses the entire file. I've tried the VS Code extension 'vscode-zpl-viewer' and with this my label works without any issues.

**Comment by roelkneepkens** [comment]
Does it work if you select the complete string to use?

The difference between this extension and the `vscode-zpl-viewer` is that:
**This extension** 
- takes part of a file to use as label (as how it is returned in API's often) --> this is where your usecase runs into issues as it is unclear for this extension where the ZPL starts and ends, hence it forces you to select the string you want to view the label for.
- supports not only raw ZPL but also base64 string containing a not only ZPL but media types like PDF, PNG etc..
 
The **vscode-zpl-viewer**
- uses the full file 
- expects always the raw ZPL.

So I guess it is a matter of determining what your use case is and use the suitable extension for it.

**Comment by rehand** [comment]
No, even if I select the content of the whole file (it's a zpl file, so the entire content should be taken) it doesn't work, so I guess some part of the content is not supported correctly.

Is the content still parsed if I select the string or is it "plain" posted to the webservice for rendering?

**Comment by bvanderhorn** [comment]
Hi @rehand, the string to select is just the ZPL string of that one label, not the entire file. Could you try it with that and let us know if it works?

The content posted to the webservice for rendering is as follows:
- If you don't select anything: the string where the cursor is in is checked left and right for _terminators_ (like '<', '>', or spaces), and the string between the terminators is sent
  - This is what did not go well in your usecase: the ZPL string itself contains '>', hence the string is cut short before sending to the webservice
- if you do select a string: the selected string is posted to the webservice

Please let us know if you have any remaining questions!
